src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -843,10 +846,13 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash {
           ScheduleTrigger(AlightVehicleTrigger(Math.max(currentLeg.endTime + 1, tick), vehicleId), self)
         )
       )
-    case Event(TriggerWithId(EndRefuelSessionTrigger(tick, _, _, vehicle), triggerId), _) =>
+    case Event(
+        TriggerWithId(EndRefuelSessionTrigger(tick, sessionDuration, fuelAddedInJoule, vehicle), triggerId),
+        _
+        ) =>
+      log.info(s"DrivesVehicle: EndRefuelSessionTrigger. tick: $tick, vehicle: $vehicle")
       if (vehicle.isConnectedToChargingPoint()) {
-        handleEndCharging(tick, vehicle)
-        vehicle.unsetParkingStall()
+        handleEndCharging(tick, vehicle, sessionDuration, fuelAddedInJoule)
```

Comment:
Should this even be doing this any longer? Or should it be delegated to the CNM? The two handling of end refueling seem very disjointed...unless I missed something. They both have an end method, but they don't seem tied together really - so the two can happen separately and alone, possibly leaving the state in a weird spot?

Reply:
This is mostly for Ridehail CAVs. This is going to be the next step, that is having Ridehail CAVs communicating with the CNM